File: engine/kirun-engine-java/src/main/java/io/kirun/engine/json/schema/validator/BooleanValidator.java
package io.kirun.engine.json.schema.validator;

import static io.kirun.engine.json.schema.validator.SchemaValidator.path;

import java.util.List;

import com.google.gson.JsonElement;
import com.google.gson.JsonPrimitive;

import io.kirun.engine.json.schema.Schema;

public class BooleanValidator {

	public static void validate(List<String> parents, Schema schema, JsonElement element) {

		if (element == null || element.isJsonNull())
			throw new SchemaValidationException(path(parents, schema.getTitle()), "Expected a boolean but found null");

		if (!element.isJsonPrimitive() || !((JsonPrimitive) element).isBoolean())
			throw new SchemaValidationException(path(parents, schema.getTitle()),
					element.toString() + " is not a boolean");
	}

	private BooleanValidator() {
	}
}

File: engine/kirun-engine-js/src/schemas/booleanValidator.js
import { path } from "./constants";

export default function booleanValidator(parents, schema, schemaRepository, functionRepository, element, validate) {
  if (!element) throw `${path(parents, schema.title)}Expected a boolean but found null`;

  if (typeof element !== "boolean") throw `${path(parents, schema.title)}${element} is not a boolean`;
}
